src/llm_vm/onsite_llm.py

```diff
@@ -527,3 +485,56 @@ def finetune(self, dataset, optimizer, c_id, small_model_filename=None):
         # optimizer.storage.set_training_in_progress(c_id, False)
         # if old_model is not None:
         #     openai.Model.delete(old_model)
+
+
+class BaseCtransformers(BaseOnsiteLLM):
+    """
+    Base Class for running Ctransformers/GGML models
+
+    Attributes:
+        model_uri (str): Hugging Face Endpoint for LLM
+        model_kwargs (dict): Keyword arguments for loading the LLM
+    
+    Methods:
+        generate: Generates a response from given prompt
+          
+    """
+    def __init__(self, model_uri=None, model_kwargs={}):
```

Comment:
Should we also have some logic for model file or just delegate it to instances ?

Reply:
the model file is passed in with `model_kwargs`

Reply:
could you explain this to me a little more? :)

Reply:
model_file needs to be a property ... right?